hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJetInstance.java

```diff
@@ -88,12 +93,14 @@
     private final JetCacheManagerImpl cacheManager;
     private final Supplier<JobRepository> jobRepository;
     private final Map<String, Observable> observables;
+    private final Map<JobListener, Map<Long, UUID>> jobListeners;
```

Comment:
There's a data race between `addJobListener()` and `JobEventInterceptor.jobStatusChanged`. The former adds an empty map, which the latter might remove before something is added to it.

Thinking about the API again, I think we can remove all the listener methods here, and only keep those in `Job` class. The methods here are essentially a utility around those in `Job`, and they aren't correctly implemented and it's not straightforward to fix. We don't need them.